### DISCUSSION TITLE
[Question] Is it a must to have the ProImageEditor as only component on screen ?

### DISCUSSION THREAD

**Discussion by Chris2137** [hm21/pro_image_editor#94] [Q&A] [answered]
### Your question

Hi Developer, 

  I tried to embed the editor into our existing application, where we have the editor in a popup dialog, and some input components on the side : 
![image](https://github.com/hm21/pro_image_editor/assets/22078631/9a59bfd8-1357-40e4-bf43-cfc41331960e)

When click the paint, it will enter edit mode (become full screen automatically), can we not go into full screen but stick with the container size and only edit within the container? 
![image](https://github.com/hm21/pro_image_editor/assets/22078631/4984cf52-1538-49c4-ae45-842aa6776e6a)

After edit, it shows correctly in the component, 
![image](https://github.com/hm21/pro_image_editor/assets/22078631/0d42d6a3-b85b-4f8e-96e2-f27b803361c5)
but if we click the paint icon again, you'll see the painting has shifted to left.
![image](https://github.com/hm21/pro_image_editor/assets/22078631/9d79b844-8fa8-4a73-8e59-8966c90cf979)

If we click OK and go back, it still shows correctly outside.

May I know is it a bug or limitation ? That we can only use the editor in full screen without other components? 

And, is it possible to have control on the Cancel and OK button to hide them? Because we want to have our own control on the dialog level to save / cancel the image editing.

Thanks!

**Comment by hm21** [comment, marked as answer]
Each subeditor, like the painting editor, opens a new page (Navigator.push(...)), so currently it's not possible to use the subeditor in the same size as the main editor.

The issue with the painting is a problem in the current version that happens when the editor size changes (as it happens when you open the subeditor). I already fixed this problem in the dev branch, because now the zero offset position of the layers is in the middle of the screen and not in the top left anymore.

To the part about “control the cancel and ok button”. Yes, it's possible to access the editor and all subeditors with a global key. To hide the buttons, you can create your own appbar that replaces the existing one, allowing you to use only the buttons you need. You can check out this [example](https://github.com/hm21/pro_image_editor/blob/stable/example/lib/pages/custom_appbar_bottombar_example.dart) how to do it.

Btw, if you have more questions, I would appreciate it if you would open a new question in the [discussions](https://github.com/hm21/pro_image_editor/discussions/categories/q-a) tab. I will also update the question template in the "Issues" tab so that users should open it there.
